src/order-book/quoteAmountsAndCostsUtils.test.ts

```diff
@@ -0,0 +1,173 @@
+import { OrderParameters, OrderKind, SigningScheme, BuyTokenDestination, SellTokenSource } from './generated'
+import { getQuoteAmountsAndCosts } from './quoteAmountsAndCostsUtils'
+
+const otherFields = {
+  buyToken: '0xdef1ca1fb7fbcdc777520aa7f396b4e015f497ab',
+  sellToken: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
+  buyTokenBalance: BuyTokenDestination.ERC20,
+  sellTokenBalance: SellTokenSource.ERC20,
+  signingScheme: SigningScheme.EIP712,
+  partiallyFillable: false,
+  receiver: '0x0000000000000000000000000000000000000000',
+  validTo: 1716904696,
+  appData: '{}',
+  appDataHash: '0x0',
+}
+
+const sellDecimals = 18
+const buyDecimals = 6
+
+/**
+ * Since we have partner fees, now it's not clear what does it mean `feeAmount`?
+ * To avoid confusion, we should consider this `feeAmount` as `networkCosts`
+ *
+ * Fee is always taken from sell token (for sell/buy orders):
+ * 3855544038281082 + 156144455961718918 = 160000000000000000
+ *
+ * Again, to avoid confusion, we should take this `sellAmount` as `sellAmountBeforeNetworkCosts`
+ * Hence, `buyAmount` is `buyAmountAfterNetworkCosts` because this amount is what you will get for the sell amount
+ *
+ * In this order we are selling 0.16 WETH for 1863 COW - network costs
+ */
+const SELL_ORDER: OrderParameters = {
+  kind: OrderKind.SELL,
+  sellAmount: '156144455961718918',
+  feeAmount: '3855544038281082',
```

Comment:
not saying we should do it, but after reading your comment, does it make sense to call also in the DTOs `feeAmount` `estimatedNetworkCosts`?

Reply:
It comes from Swagger sheme, I think we can change it in API only with a new version of the API

Reply:
okay, they will never change it in the API, as it is part of the protocol (the field is feeAmount)

Although is silly, because now we post the fee with zero, so I think our library should abstract all these things to the user and give them exactly all the info with nace names, and even the order details to sign